integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/testsupport/types/AbstractFieldTypeDescriptor.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.integrationtest.backend.tck.testsupport.types;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import org.hibernate.search.engine.backend.types.dsl.IndexFieldTypeFactory;
+import org.hibernate.search.engine.backend.types.dsl.IndexFieldTypeOptionsStep;
+import org.hibernate.search.integrationtest.backend.tck.testsupport.types.expectations.IndexNullAsMatchPredicateExpectactions;
+
+public abstract class AbstractFieldTypeDescriptor<F> {
+
+	protected final Class<F> javaType;
+	protected final String uniqueName;
+
+	private final List<F> uniquelyMatchableValues = Collections.unmodifiableList( createUniquelyMatchableValues() );
+
+	private final List<F> nonMatchingValues = Collections.unmodifiableList( createNonMatchingValues() );
+
+	protected AbstractFieldTypeDescriptor(Class<F> javaType) {
+		this( javaType, javaType.getSimpleName() );
+	}
+
+	protected AbstractFieldTypeDescriptor(Class<F> javaType, String uniqueName) {
+		this.javaType = javaType;
+		this.uniqueName = uniqueName;
+	}
+
+	@Override
+	public String toString() {
+		return getUniqueName();
+	}
+
+	public final Class<F> getJavaType() {
+		return javaType;
+	}
+
+	public final String getUniqueName() {
+		return uniqueName;
+	}
+
+	public IndexFieldTypeOptionsStep<?, F> configure(IndexFieldTypeFactory fieldContext) {
```

Comment:
I was trying to get vectors to fit this pattern for tests, but since vectors and other fields have a different base options step the most common one is this `IndexFieldTypeOptionsStep` and if I change the configure method to it then there's a looooot of tests that are getting affected and needs some casts 😔 So I'm not sure it is worth to fit the vectors into these regular fields tests...

Reply:
I think you probably could get by with some generics? E.g. something like https://github.com/marko-bekhta/hibernate-search/pull/212

Note if you need to add a superinterface common to both `StandardIndexFieldTypeOptionsStep` and `VectorIndexFieldTypeOptionsStep`, that's fine by me. The only problem being the name...